File: .Tests/CyclomaticComplexity.cs
public class CyclomaticComplexity
{
    public static int CalculateCyclomaticComplexity(string code)
    {
        int complexity = 1;  // Мінімальна складність = 1 (для одного шляху)

        string[] decisionPoints = { "if", "for", "while", "foreach", "case", "catch", "try", "else" };

        foreach (var decision in decisionPoints)
        {
            complexity += CountOccurrences(code, decision);
        }

        return complexity;
    }

    private static int CountOccurrences(string code, string decisionPoint)
    {
        int count = 0;
        int index = 0;

        while ((index = code.IndexOf(decisionPoint, index)) != -1)
        {
            count++;
            index += decisionPoint.Length;
        }

        return count;
    }
}

File: .Tests/cyclomatic_complexity.py
def calculate_cyclomatic_complexity(code):
    """
    Функція для розрахунку цикломатичної складності програмного коду.
    Параметри:
    code (str): Код, який потрібно проаналізувати.

    Повертає:
    int: Цикломатична складність.
    """
    conditions = code.count('if') + code.count('while') + code.count('for')
    edges = code.count('else') + code.count('elif')
    vertices = code.count('{') + code.count('}')
    
    return conditions + edges - vertices + 2  # Формула МакКейба

# Приклад коду
sample_code = """
if x > 10:
    for i in range(5):
        print(i)
else:
    print("x is less than 10")
"""

complexity = calculate_cyclomatic_complexity(sample_code)
print("Цикломатична складність:", complexity)
